src/vm_core.rs

```diff
@@ -152,6 +152,17 @@ impl PyVM {
             Err(e) => Err(e),
         }
     }
+
+    pub(crate) fn prepare_os_context(&self) -> Vec<MaybeRelocatable> {
```

Comment:
I'm not sure if I remember it correctly, but the reason it's `MaybeRelocatable` is some may be relocated already when accessed, right? Otherwise, since these are all pointers we could just use `Relocatable`.